### ISSUE TITLE
Internal/Bridge switching is a bit unintuitve.

### ISSUE THREAD

**Issue by tomusdrw** [paritytech/parity-bridges-ui#278]
IMHO it would be much better to have "Token" selector instead of separating Internal/Bridge transfers. Currently switching between these two completely wipes out all of the details entered. So if user realises they've made a mistake (i.e. wanted to transfer different tokens) they have to start from scratch.

This is especially important for the future in case we are going to support multiple different tokens that can exist on one network and multiple networks.

So basically the flow would be:
1. Selecting "Sender" defines the Source network.
2. Selecting "Token" defines the Target network (Which might be the same as Source network).

**Comment by goldsteinsveta** [comment]
A little prototype here:
cc @hbulgarini 
https://user-images.githubusercontent.com/8782596/132690095-30024ab1-6ba7-4a28-ad3d-0cfe5d8833a3.mov

**Comment by hbulgarini** [comment]
Thanks @goldsteinsveta for the prototype.

I understand what @tomusdrw is proposing but in this case I have to disagree that for the user the flow proposed for switching between local/bridge will be more intuitive. Even though switching between local/bridge now has the downside of facing the form reseted and having to do it all from scratch, I think having that switch on the top is the easiest way to understand what kind of transaction the user will execute.  

I see two main issues with this design:

1) Choosing the type of transaction the user wants to execute is the first action that needs to be decided and this action is not visible until you click the token section, which is kind of hidden at that time. I feel that the user is forced to have previous knowledge if wants to do the switch.

2) If the user reach the token dropdown, there is an extra effort to understand that choosing the same token as the source means local transfer ( even though we put an explanatory text for clarifying this )

thoughts?

1) Choosing the type of transaction the user wants to execute is the first action that needs to be decided and this action is not visible until you click the token section. The user is forced to have a previous knowledge about how to do this.
2) If the user reach the token dropdown, there is an extra effort to understand that choosing the same token as the source means local transfer ( even though we put an explanatory text for clarifying this )

**Comment by tomusdrw** [comment]
@goldsteinsveta I do like it, I'm curious how it would look like with more tokens to choose from though.

@hbulgarini 

> Choosing the type of transaction the user wants to execute is the first action that needs to be decided

No, I don't think so. If that was the first action, then the decision would be: I'm either using bridge UI or some other "local UI". The whole point of designing the UX around "Sender account" is that the flow is reversed - you don't decide what networks you want to be connected to, where the transfer happens, or any other technical details, but rather the UI tells you what you can do with the account you've selected (transfer SRC tokens, TAR tokens, or any other tokens that you hold on any other target network).

> If the user reach the token dropdown, there is an extra effort to understand that choosing the same token as the source means local transfer ( even though we put an explanatory text for clarifying this )

Why would the user care if it's local or bridge? The user want's to spend tokens "X" that he/she hold in account "A" - the fact that it's going to be local/bridge transfer is irrelevant to their intention (i.e. there is no other way for them to spend these tokens). Sure, under the hood it's different transaction and different fees, but if you focus on the user intention it really doesn't matter. You should put the user and the sender account in the center of your thinking: "Given this source account - what can I do?" With multiple tokens support on many networks, the token-network association gets blurry, and for sure the distinction between "native tokens" and "other assets".

**Comment by hbulgarini** [comment]
> you don't decide what networks you want to be connected to, where the transfer happens, or any other technical details, but rather the UI tells you what you can do with the account you've selected

I understand your point about having a super smart UX that leads the user to a simple path. But me being a really careful user, the lack of involvement in the steps needed to achieve the goal i want to take , would make me feel worried because i would "loose the control" ( completely personal based on my day to day experience )

Also i'm concerned that a super smart app would force the user to invest more energy in understanding what it is going on under the hood and what action the user is executing. That's why having a simple switch that says: Local/Bridge is a really easy way to understand. 

Let's suppose we remove the local/bridge button and the user switches between the token dropdown list making the local/bridge to switch. The result is that instead of having a fee of 0.3SRC + 5 TAR, the calculated fee will switch to only 0.03 SRC. My guess is that the user will be force to understand why that fee changed happen where if the user is there indicating that the transfer type is local/bridge it will be more clear.

@goldsteinsveta , do you have any point of view regarding this discussion? i wonder if giving so much power to the user by "unsmarting" the UX is what the regular user feels as "complex" &#129300;

**Comment by tomusdrw** [comment]
This is not a very good venue to discuss product decisions, but let me try to address some of your points.

>  But me being a really careful user, the lack of involvement in the steps needed to achieve the goal i want to take ,

Then you are clearly not the target user of the application we're building here. It's not about the app being "smart", it's about the user, who doesn't care about this stuff. The intention of the user is to transfer token X that he/she holds in account A to received address B. The UI is supposed to take the burden from them of determining: "What kind of transfer is it going to be?",  "Which interface I should be using?", etc.

> My guess is that the user will be force to understand why that fee changed happen where if the user is there indicating that the transfer type is local/bridge it will be more clear.

That's the point of the UI to make sure that whatever is going to happen is clearly communicated. You are in a privileged position  to having work on the subject for quite a while so your understanding of the topic is much deeper than for the regular user. I.e. you not only know what you want to do, but also how it will be done. We shouldn't be assuming that the user reads a shit ton of documentation BEFORE using the UI, so that they are knowledgeable enough to be able to understand HOW to achieve WHAT they want (the intention). Please assume the user comes with "WHAT" and the UI is there to tell them "HOW" and explain in the best possible way "WHY" and what costs the user will need to pay.

**Comment by hbulgarini** [comment]
Right. As soon as we extend the use cases on tokens/parachains/chains this will become more difficult to make it explicit and it will be harder for the user to understand it. We should abstract all this logic and make it transparent for the user.
